shared/js/inject/chrome.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Inject all the overwrites into the page.
+ */
+function inject (code) {
+    const elem = document.head || document.documentElement
+    // Inject into main page
+    try {
+        const e = document.createElement('script')
+        e.textContent = `(() => {
+            ${code}
+        })();`
+        elem.appendChild(e)
+        e.remove()
+    } catch (e) {
+    }
+}
+
+inject()
+
+function randomString () {
+    const num = crypto.getRandomValues(new Uint32Array(1))[0] / 2 ** 32
+    return num.toString().replace('0.', '')
+}
+
+function init () {
+    const randomMethodName = '_d' + randomString()
+    const randomPassword = '_p' + randomString()
+    const randomFunction = `
+      // Use define property so isn't enumerable
+      Object.defineProperty(window, '${randomMethodName}', {
+          enumerable: false,
+          // configurable, To allow for deletion later
+          configurable: true,
+          writable: false,
+          // Use proxy to ensure stringification isn't possible
+          value: new Proxy(function () {}, {
+              apply(target, thisArg, args) {
+                  if ('${randomPassword}' === args[0]) {
+                      protections.initProtection(args[1])
+                  } else {
+                      // TODO force enable all protections if password is wrong
```

Comment:
@sammacbeth I think this is reasonable to apply protections when a page has tampered with our code right?

Reply:
IMO it depends if it would be possible for them to prevent our valid call to this function.

Reply:
I'm pretty sure they can given timing attacks, I think the risk here is low to always force the protections on right?